src/include/txn.i

```diff
@@ -765,22 +765,13 @@ __upd_alloc_tombstone(
 }
 
 /*
- * __wt_txn_read --
+ * __wt_txn_read_upd_list --
```

Comment:
This change is not strictly related. But my change exposed another visibility issue so I figured I should fix it and keep the build green since it isn't hard to do.

Essentially, when we call `__wt_txn_read` to search an insert list, we're not intending to check the onpage value or the history store as there shouldn't be any content there anyway. So I've extracted out the iteration of the update list into a separate function and used that where appropriate.

Reply:
Maybe this should be `__wt_txn_read_ins_list` then? There are update lists that aren't insert lists?

Reply:
I use this in `__wt_txn_read` to read the update list on the modify structure (not insert list). So I don't think I can rename this to something more specific.